### PR TITLE
Support One-Versus-One approach

### PR DESCRIPTION
**Pull request by FedyuninV**

### CODE REVIEW

File: NeoML/docs/ru/API/ClassificationAndRegression/OneVersusOne.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	- [Результат классификации](#результат-классификации)
+	- [Пример](#пример)
+
+<!-- /TOC -->
+
+Данный метод позволяет выполнить многоклассовую классификацию, имея только бинарный классификатор.
+
+Метод заключается в преобразовании исходной многоклассовой задачи в набор бинарных задач для каждой пары классов. На этом наборе обучается соответствующее количество бинарных классификаторов.
+
+Затем, вероятности принадлжености объектов к каждому из классов ищутся путем решения оптимизационной задачи, описанной в Section 4 [данной статьи](https://www.csie.ntu.edu.tw/~cjlin/papers/svmprob/svmprob.pdf).
```
Comment (SAngeliuk) [inline NeoML/docs/ru/API/ClassificationAndRegression/OneVersusOne.md]: ринадлжености

File: NeoML/src/TraditionalML/OneVersusOneModel.cpp [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+		if( diff > eps ) {
+			return false;
+		}
+	}
+	return true;
+}
+
+// fills prob with approximated probabilities based on pred
+// pred is a matrix of size classCount x classCount (r in the article)
+void findProb( const CArray<CArray<float>>& pred, CArray<float>& prob )
```
Comment (SAngeliuk) [inline NeoML/src/TraditionalML/OneVersusOneModel.cpp]: static

File: NeoML/docs/ru/API/ClassificationAndRegression/Linear.md [hunk 3]
```diff
@@ -27,7 +27,8 @@
 - *Tolerance* — критерий останова;
 - *NormalizeError* — указывает, необходима ли нормализация ошибки;
 - *L1Coeff* — коэффициент L1 регуляризации; чтобы использовать L2-регуляризацию, присвойте ему значение `0`;
-- *ThreadCount* — количество потоков, которое можно использовать во время обучения.
+- *ThreadCount* — количество потоков, которое можно использовать во время обучения;
+- *MulticlassMode* - подход, используемый при многоклассовой классификации.
```
Comment (SAngeliuk) [inline NeoML/docs/ru/API/ClassificationAndRegression/Linear.md]: Nothing about one vs one model...

File: NeoML/docs/ru/API/ClassificationAndRegression/DecisionTree.md [hunk 4]
```diff
@@ -26,7 +26,8 @@
 - *MaxNodesCount* — максимальное количество вершин в дереве;
 - *SplitCriterion* — критерий деления подмножеств при построении дерева;
 - *ConstNodeThreshold* — доля одинаковых элементов в подмножестве, при превышении которой будет создана константная вершина (может принимать значения от 0 до 1);
-- *RandomSelectedFeaturesCount* — при построении каждого узла используется не больше этого количества случайно выбранных признаков. Задайте значение `-1`, чтобы использовать все признаки.
+- *RandomSelectedFeaturesCount* — при построении каждого узла используется не больше этого количества случайно выбранных признаков. Задайте значение `-1`, чтобы использовать все признаки;
+- *MulticlassMode* - подход, используемый при многоклассовой классификации.
 
 ## Модель
```
Comment (SAngeliuk) [inline NeoML/docs/ru/API/ClassificationAndRegression/DecisionTree.md:32]: Nothing about one vs one model...

File: NeoML/docs/en/API/ClassificationAndRegression/Linear.md [hunk 5]
```diff
@@ -28,6 +28,7 @@ The parameters are represented by a `CLinear::CParams` structure.
 - *NormalizeError* — specifies if the error should be normalized.
 - *L1Coeff* — the L1 regularization coefficient; set to `0` to use the L2 regularization instead.
 - *ThreadCount* — the number of processing threads to be used while training the model.
+- *MulticlassMode* - the approach used in multiclass task.
```
Comment (SAngeliuk) [inline NeoML/docs/en/API/ClassificationAndRegression/Linear.md]: Nothing about one vs one model...

File: NeoML/docs/en/API/ClassificationAndRegression/DecisionTree.md [hunk 6]
```diff
@@ -27,6 +27,7 @@ The parameters are represented by a `CDecisionTree::CParams` structure.
 - *SplitCriterion* — the criterion for subset splitting.
 - *ConstNodeThreshold* — the ratio of the equal elements in the subset which should be the threshold for creating a constant node (may be from 0 to 1).
 - *RandomSelectedFeaturesCount* — no more than this number of randomly selected features will be used for each node. Set the value to `-1` to use all features every time.
+- *MulticlassMode* - the approach used in multiclass task.
```
Comment (SAngeliuk) [inline NeoML/docs/en/API/ClassificationAndRegression/DecisionTree.md]: Nothing about one vs one model...

Comment (FedyuninV) [reply NeoML/docs/en/API/ClassificationAndRegression/DecisionTree.md]: Fxied

File: NeoML/docs/en/API/ClassificationAndRegression/Linear.md (hunk 5 above)
Comment (FedyuninV) [reply NeoML/docs/en/API/ClassificationAndRegression/Linear.md]: Fxied

File: NeoML/docs/ru/API/ClassificationAndRegression/DecisionTree.md (hunk 4 above)
Comment (FedyuninV) [reply NeoML/docs/ru/API/ClassificationAndRegression/DecisionTree.md:32]: Fxied

File: NeoML/docs/ru/API/ClassificationAndRegression/Linear.md (hunk 3 above)
Comment (FedyuninV) [reply NeoML/docs/ru/API/ClassificationAndRegression/Linear.md]: Fxied

File: NeoML/src/TraditionalML/OneVersusOneModel.cpp (hunk 2 above)
Comment (FedyuninV) [reply NeoML/src/TraditionalML/OneVersusOneModel.cpp]: Fxied

File: NeoML/docs/ru/API/ClassificationAndRegression/OneVersusOne.md (hunk 1 above)
Comment (FedyuninV) [reply NeoML/docs/ru/API/ClassificationAndRegression/OneVersusOne.md]: Fxied

**Review by SAngeliuk — APPROVED** [review]

**Pull request merged**